src/Components/Components/src/Routing/RegexConstraintSupport.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Routing;
+
+internal static class RegexConstraintSupport
+{
+    // We should check the AppContext switch in the implementation, but it doesn't flow to the wasm runtime
+    // during development, so we can't offer a better experience (build time message to enable the switch)
+    // until the context switch flows to the runtime.
```

Comment:
There is no AppContext switch support for wasm, the issue tracking it is https://github.com/dotnet/runtime/issues/97449

Reply:
> you only get notified at publish time

Does that mean "while `dotnet publish` is executing, and will fail the build otherwise", or does it mean "at runtime in production"?

Reply:
Actually, I think we can still achieve the experience that we want, we just need to flow the runtime config option during development via a separate mechanism, which I think we can do through an assembly level attribute.

The main current challenge is that the value is not flowed through the standard machinery, but that's something we can do ourselves on the build.

I'll prototype that so that we can backport this to 8.0, even if for 9.0 we'll rely on the runtime feature once implemented, this is hopefully a 1 off.

Reply:
I'm fine with if you want to do that, but am not sure there's a requirement to patch 8.0 here. We didn't make any claim that this works on 8.0 and AFAIK there hasn't been any customer feedback about it. Totally up to you.

Reply:
going to fix this in main, and we can decide whether to backport it later